### PR TITLE
[Ameba] Implement reading and decoding factorydata from flash

### PR DESCRIPTION
**Pull request by pankore**
- Ameba implementation of reading and decoding factorydata from flash
- Add FactoryDataDecoder
- Revise FactoryDataProvider to use factorydata from flash
- Change chipinterface initialization order
- Add CHIP_ENABLE_AMEBA_FACTORY_DATA macro to enable/disable flash factorydata
- Revise filepath in mbedtls.cmake
- Requires docker update first

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]

**Comment by andy31415** [comment]
@pankore it looks like we have an ameba failure on this PR.

File: third_party/ameba/mbedtls.cmake [hunk 1]
```diff
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.6)
 
 project(mbedtls)
 
-set(dir "${sdk_root}/component/common/network/ssl/mbedtls-matter")
+set(dir "${sdk_root}/component/common/application/matter/mbedtls")
```
Comment (andy31415) [inline third_party/ameba/mbedtls.cmake:5]: does this depend on some updated SDK? We get build failures it seems.

Comment (pankore) [reply third_party/ameba/mbedtls.cmake:5]: @andy31415 , yes there is a docker update #23759 that needs to be merged first, but that PR also has some merge conflict, we will fix it

Comment (andy31415) [reply third_party/ameba/mbedtls.cmake]: Ameba build is still failing. I merged with master to see if that makes it work better.

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
Changes are in Ameba only and ameba compilation passed. pullapprove seems stuck ("given usersnames could not be requested") so I will bypass CI checks and merge this.

**Pull request merged**